lib/marked.js

```diff
@@ -539,7 +539,7 @@ inline.pedantic = merge({}, inline.normal, {
 
 inline.gfm = merge({}, inline.normal, {
   escape: replace(inline.escape)('])', '~|])')(),
-  url: XRegExp.cache('^((?:[A-Za-z][\\pL\\d-]+:(?:\\/{1,3}|[\\pL\\d%])|www\\d{0,3}[.]|[\\pL\\d.\\-]+[.]\\pL{2,4}\\/)(?:\\[[\\da-f:]+\\]|[^\\s`!()\\[\\]{};:\'",<>?«»“”‘’]|[`!\\[\\]{};:\'",<>?«»“”‘’](?=[^\\s()<>])|\\((?:[^\\s()<>]|(?:\\([^\\s()<>]+\\)))*\\))+)', ''),
+  url: XRegExp.cache('^((?:[A-Za-z][\\pL\\d-]+([.+][A-Za-z\\d]+)?:(?:\\/{1,3}|[\\pL\\d%])|www\\d{0,3}[.]|[\\pL\\d.\\-]+[.]\\pL{2,4}\\/)(?:\\[[\\da-f:]+\\]|[^\\s`!()\\[\\]{};:\'",<>?«»“”‘’]|[`!\\[\\]{};:\'",<>?«»“”‘’](?=[^\\s()<>])|\\((?:[^\\s()<>]|(?:\\([^\\s()<>]+\\)))*\\))+)', ''),
```

Comment:
>>I'm curious, to start, why [A-Za-z] was there at all -- seems like any valid Unicode letter should be allowed at the start of the scheme?
The `[A-Z`part was introduced from a previous PR to allow uppercase alphabet characters that weren't supported before. However I was also wondering if `a-z]` was useful at all regarding the presence of `\\pL`.

>> I'm not sure this exactly works ...
If `taco++......++what` is a correct scheme then your proposition is correct but should include the uppercase case. I don't know what the RFC tells about the repetition of `+`,  `-` or `.`. The regex I implemented just want the spec in the issue to be validated `taco+what` or `taco.what` or the previous scheme.

Reply:
Good point for the RFC. However, the RFC schemes requires `alpha` as a starting character and any repetition of alpha, digit, plus, dash, or dot for the following character so `\\pL` won't work because it will match foreign languages as well.

`````
      alpha    = lowalpha | upalpha

      lowalpha = "a" | "b" | "c" | "d" | "e" | "f" | "g" | "h" | "i" |
                 "j" | "k" | "l" | "m" | "n" | "o" | "p" | "q" | "r" |
                 "s" | "t" | "u" | "v" | "w" | "x" | "y" | "z"

      upalpha  = "A" | "B" | "C" | "D" | "E" | "F" | "G" | "H" | "I" |
                 "J" | "K" | "L" | "M" | "N" | "O" | "P" | "Q" | "R" |
                 "S" | "T" | "U" | "V" | "W" | "X" | "Y" | "Z"
`````
Hence I think that the correct regex would be `[A-Za-z][A-Za-z\\d.+-]*`

Reply:
Good catch that the scheme should only Latin letters. I think I got a bit overly enthusiastic with putting `\pL` everywhere when I was adding support for non-Latin characters in the TLD. I'm all for the simpler regex for that part as well